pandas/io/parsers.py

```diff
@@ -1620,6 +1620,12 @@ def __init__(self, src, **kwds):
 
         if self.usecols:
             usecols = _evaluate_usecols(self.usecols, self.orig_names)
+
+            #gh-14671
```

Comment:
`_evaluate_usecols` is purely for evaluation.  I don't think it was meant to have validation like this inside its implementation as well.